src/SourceGenerators/Uno.UI.SourceGenerators.Internal/DependencyObject/DependencyPropertyGenerator.cs

```diff
@@ -2,530 +2,589 @@
 
-				changedCallbackName ??= $"On{propertyName}Changed";
-				var propertyChangedMethods = propertyOwnerType.GetMethodsWithName(changedCallbackName).ToArray();
-				if (changedCallback || (propertyChangedMethods?.Any() ?? false))
+				var changedCallback = GetBooleanAttributeValue(attribute, "ChangedCallback", false);
+				var propertyChangedMethods = dpSymbol.ContainingType.GetMethodsWithName(ChangedCallbackName).ToArray();
+				if (changedCallback || propertyChangedMethods.Length > 0)
 				{
-					if (propertyChangedMethods.FirstOrDefault(IsCallbackWithDPChangedArgs) is { } callbackWithEventArgs)
+					if (propertyChangedMethods.Any(m => IsCallbackWithDPChangedArgs(m)))
 					{
-						builder.AppendLineIndented($"\t\t, propertyChangedCallback: (instance, args) => {changedCallbackName}(instance, args)");
+						IsCallbackWithDPChangedArgs = true;
 					}
-					else if (propertyChangedMethods.FirstOrDefault(IsCallbackWithDPChangedArgsOnly) is { } callbackWithEventArgsOnly)
+					else if (propertyChangedMethods.Any(m => IsCallbackWithDPChangedArgsOnly(m)))
 					{
-						builder.AppendLineIndented($"\t\t, propertyChangedCallback: (instance, args) => {changedCallbackName}(args)");
+						IsCallbackWithDPChangedArgsOnly = true;
 					}
-					else if (propertyChangedMethods?.FirstOrDefault(m => m?.Parameters.Length == 2) is { } callbackWithOldAndNew)
+					else if (propertyChangedMethods.Any(m => m.Parameters.Length == 2))
 					{
-						builder.AppendLineIndented($"\t\t, propertyChangedCallback: (instance, args) => {changedCallbackName}(({propertyTypeName})args.OldValue, ({propertyTypeName})args.NewValue)");
+						CallBackWithOldAndNew = true;
 					}
 					else
 					{
-						builder.AppendLineIndented($"#error Valid {changedCallbackName} not found.  Must be {changedCallbackName}(DependencyPropertyChangedEventArgs), {changedCallbackName}(Instance, DependencyPropertyChangedEventArgs) or {changedCallbackName}(oldValue, newValue)");
+						IsInvalidChangedCallbackName = true;
 					}
 				}
 
-				builder.AppendLineIndented($"));");
-
-				builder.AppendLineIndented($"#endregion");
+				if (IsAttached)
+				{
+					AttachedPropertyData = new AttachedPropertyData(dpSymbol, attribute, PropertyName);
+				}
+				else
+				{
+					PropertyData = new PropertyData(dpSymbol, PropertyName);
+				}
 			}
+		}
 
-			private bool IsCallbackWithDPChangedArgsOnly(IMethodSymbol m)
-				=> SymbolEqualityComparer.Default.Equals(m?.Parameters.FirstOrDefault()?.Type, _dependencyPropertyChangedEventArgsSymbol);
+		public void Initialize(IncrementalGeneratorInitializationContext context)
+		{
+			var attributedSymbolsProvider = context.SyntaxProvider.ForAttributeWithMetadataName(
+				"Uno.UI.Xaml.GeneratedDependencyPropertyAttribute",
+				static (node, _) => node.IsKind(SyntaxKind.PropertyDeclaration),
+				static (context, token) =>
+				{
+					var attribute = context.Attributes[0];
+					return new GenerationCandidateData(context.TargetSymbol, attribute);
```

Comment:
Do you know how roslyn compute the equality for `GenerationCandidateData`? Is it done through reflection?

Reply:
I'm not sure. But it's quite possible that structs are compared through reflection by default.

Let me use a `record`